osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/AttributeAddressHelperService.java

```diff
@@ -0,0 +1,248 @@
+    private static final byte ATTRIBUTE_ID_BUFFER = 2;
+    private static final byte ATTRIBUTE_ID_SCALER_UNIT = 3;
+    private static final ObisCode OBIS_CODE_INTERVAL_IMPORT_SCALER_UNIT = new ObisCode("1.0.1.8.0.255");
+    private static final ObisCode OBIS_CODE_INTERVAL_EXPORT_SCALER_UNIT = new ObisCode("1.0.2.8.0.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_IMPORT_RATE_1_SCALER_UNIT = new ObisCode("1.0.1.8.1.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_IMPORT_RATE_2_SCALER_UNIT = new ObisCode("1.0.1.8.2.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_EXPORT_RATE_1_SCALER_UNIT = new ObisCode("1.0.2.8.1.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_EXPORT_RATE_2_SCALER_UNIT = new ObisCode("1.0.2.8.2.255");
+
+    private static final int CLASS_ID_REGISTER = 3;
+
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_IMPORT_RATE_1 = new byte[] { 1, 0, 1, 8, 1, (byte) 255 };
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_IMPORT_RATE_2 = new byte[] { 1, 0, 1, 8, 2, (byte) 255 };
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_EXPORT_RATE_1 = new byte[] { 1, 0, 2, 8, 1, (byte) 255 };
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_EXPORT_RATE_2 = new byte[] { 1, 0, 2, 8, 2, (byte) 255 };
+    private static final byte ATTRIBUTE_ID_VALUE = 2;
+
+    private static final int ACCESS_SELECTOR_RANGE_DESCRIPTOR = 1;
+
+    private final DlmsHelperService dlmsHelperService;
+
+    @Autowired
+    public AttributeAddressHelperService(final DlmsHelperService dlmsHelperService) {
+        this.dlmsHelperService = dlmsHelperService;
+    }
+
+    public AttributeAddress[] getProfileBufferAndScalerUnitForPeriodicMeterReads(final PeriodTypeDto periodType,
+            final DateTime beginDateTime, final DateTime endDateTime, final boolean isSelectingValuesSupported)
+            throws ProtocolAdapterException {
+
+        final SelectiveAccessDescription access = this.getSelectiveAccessDescription(periodType, beginDateTime,
+                endDateTime, isSelectingValuesSupported);
+
+        final List<AttributeAddress> profileBuffer = new ArrayList<>();
+        switch (periodType) {
+        case INTERVAL:
+            profileBuffer.add(new AttributeAddress(CLASS_ID_PROFILE_GENERIC, OBIS_CODE_INTERVAL_BILLING,
+                    ATTRIBUTE_ID_BUFFER, access));
+            break;
+        case DAILY:
+            profileBuffer.add(new AttributeAddress(CLASS_ID_PROFILE_GENERIC, OBIS_CODE_DAILY_BILLING,
+                    ATTRIBUTE_ID_BUFFER, access));
+            break;
+        case MONTHLY:
+            profileBuffer.add(new AttributeAddress(CLASS_ID_PROFILE_GENERIC, OBIS_CODE_MONTHLY_BILLING,
+                    ATTRIBUTE_ID_BUFFER, access));
+            break;
+        default:
+            throw new ProtocolAdapterException(String.format("periodtype %s not supported", periodType));
+        }
```

Comment:
Extract this switch into a separate method with an appropriate name, let's say 'getProfileBuffer'.

Reply:
Aangepast